server/backend/db_backend.go

```diff
@@ -605,6 +609,47 @@ func (self *MongoBackend) updateContract(contract *models.Contract) bool {
 	return true
 }
 
+func (self *MongoBackend) getContracts(filter *models.ContractsFilter) []*models.Address {
+	var addresses []*models.Address
+	var sortQuery string
+	findQuery := bson.M{"contract": true}
+	if filter.TokenName != "" {
+		findQuery["token_name"] = bson.RegEx{regexp.QuoteMeta(filter.TokenName), "i"}
+	}
+	if filter.TokenSymbol != "" {
+		findQuery["token_symbol"] = bson.RegEx{regexp.QuoteMeta(filter.TokenSymbol), "i"}
+	}
+	if filter.ErcType != "" {
+		println(filter.ErcType)
+		findQuery["erc_types"] = filter.ErcType
+	}
+	if filter.SortBy != "" {
+		sortQuery = filter.SortBy
+		if filter.Asc == false {
+			sortQuery = "-" + sortQuery
+		}
+	} else {
+		sortQuery = "-number_of_token_holders"
+	}
+	if filter.Skip < 0 {
+		filter.Skip = defaultSkip
+	}
+	if filter.Limit < 0 || filter.Limit > defaultFetchLimit {
+		filter.Limit = defaultFetchLimit
+	}
+	err := self.mongo.
+		C("Addresses").
+		Find(findQuery).
+		Sort(sortQuery).
+		Skip(filter.Skip).
+		Limit(filter.Limit).
+		All(&addresses)
+	if err != nil {
+		log.Debug().Err(err).Msg("GetContracts")
```

Comment:
This is the trouble. Need to pass in the new logger type, and this is probably ERROR level, isn't it?
```suggestion
		lgr.Error("Failed to query contracts", zap.Error(err))
```

Reply:
@jmank88  yes, fixed